### PR TITLE
add onChange method

### LINKED ISSUE #36: Add onChange method
Update to have the `onChange` method

### PR DESCRIPTION
**Pull request by grcldq**
fixes #36

### CODE REVIEW

File: src/form.ts [hunk 1]
```diff
@@ -11,6 +11,7 @@ export class Form<T> {
   #validations: MappedValidation<T>;
   #onUpdate?: () => void;
   #field: FieldImplementation<T, T>;
+  #onChange?: (model: T) => void;
```
Comment (JensRavens) [inline src/form.ts:14]:
```suggestion
  #onChange?: (form: Form<T>) => void;
```

this way you can get the model or just the changes or any other state about the form.

**Review by JensRavens — CHANGES_REQUESTED** [review]

**Review by JensRavens — CHANGES_REQUESTED** [review]
what I'm missing on this PR is some test coverage for the new method

**Comment by grcldq** [comment]
> what I'm missing on this PR is some test coverage for the new method

added 2 tests for this, please check again.

**Review by JensRavens — APPROVED** [review]

**Review by JensRavens — COMMENTED** [review]
I found the following issues:
- font rendering still seems to be wrong (not blocking, but something we should look into)
- when typing the investment amount field, the bottom border becomes thicker when the first number is entered. If this is the active/focus state of this field, it should happen when the cursor enters/leaves, not depending on the content
- when I type 80.000 into the field, it is marked red without a validation error displayed
- the bottom border of the date input is always bold, no matter if it is active or not
- the placeholder "nächstmöglich" in the datefield should display a text cursor mouse cursor to indicate that you can edit it
- spacings around the dynamization slider are incorrect, the handle of the slider is bigger than in the spec

**Pull request merged**